packages/manager/src/features/Lish/Weblish.tsx

```diff
@@ -56,13 +62,59 @@ export class Weblish extends React.Component<Props, State> {
   connect() {
     const { weblish_url, ws_protocols } = this.props;
 
-    this.socket = new WebSocket(weblish_url, ws_protocols);
+    /* When this.socket != origSocket, the socket from this connect()
+     * call has been closed and possibly replaced by a new socket. */
+    const origSocket = new WebSocket(weblish_url, ws_protocols);
+    this.socket = origSocket;
+
+    this.lastMessage = '';
+    this.setState({error: ''});
 
     this.socket.addEventListener('open', () => {
       if (!this.mounted) {
         return;
       }
-      this.setState({ renderingLish: true }, () => this.renderTerminal());
+      this.setState({ renderingLish: true }, () => this.renderTerminal(origSocket));
+    });
+
+    this.socket.addEventListener('close', (evt) => {
+      if (this.socket != origSocket) {
```

Comment:
We use double not equal over single not equal for type safety
```suggestion
      if (this.socket !== origSocket) {
```

Reply:
Done!